mm2src/mm2_main/src/lp_swap.rs

```diff
@@ -1299,68 +1373,85 @@ pub async fn latest_swaps_for_pair(
+                    Ok(None) => warn!("No such swap with the uuid '{}'", uuid),
+                    Err(e) => error!(
+                        "Error loading a swap with the uuid '{}': {} for db_id=({db_id})",
+                        uuid, e
+                    ),
                 },
-                Ok(None) => warn!("No such swap with the uuid '{}'", uuid),
-                Err(e) => error!("Error loading a swap with the uuid '{}': {}", uuid, e),
-            },
-            MAKER_SWAP_V2_TYPE => match get_maker_swap_data_for_rpc(&ctx, uuid).await {
-                Ok(data) => {
-                    let swap_json = try_s!(json::to_value(data));
-                    swaps.push(swap_json);
+                MAKER_SWAP_V2_TYPE => match get_maker_swap_data_for_rpc(&ctx, uuid, Some(&db_result.pubkey)).await {
+                    Ok(data) => {
+                        let swap_json = try_s!(json::to_value(data));
+                        swaps.push(swap_json);
+                    },
+                    Err(e) => error!(
+                        "Error loading a swap with the uuid '{}': {} for db_id=({db_id})",
+                        uuid, e
+                    ),
                 },
-                Err(e) => error!("Error loading a swap with the uuid '{}': {}", uuid, e),
-            },
-            TAKER_SWAP_V2_TYPE => match get_taker_swap_data_for_rpc(&ctx, uuid).await {
-                Ok(data) => {
-                    let swap_json = try_s!(json::to_value(data));
-                    swaps.push(swap_json);
+                TAKER_SWAP_V2_TYPE => match get_taker_swap_data_for_rpc(&ctx, uuid, Some(&db_result.pubkey)).await {
+                    Ok(data) => {
+                        let swap_json = try_s!(json::to_value(data));
+                        swaps.push(swap_json);
+                    },
+                    Err(e) => error!(
+                        "Error loading a swap with the uuid '{}': {} for db_id=({db_id})",
+                        uuid, e
+                    ),
                 },
-                Err(e) => error!("Error loading a swap with the uuid '{}': {}", uuid, e),
-            },
-            unknown_type => error!("Swap with the uuid '{}' has unknown type {}", uuid, unknown_type),
+                unknown_type => error!("Swap with the uuid '{}' has unknown type {}", uuid, unknown_type),
+            }
+        }
+
+        if !swaps.is_empty() {
+            res_js.push(json!({
+                "swaps": swaps,
+                "from_uuid": req.paging_options.from_uuid,
+                "skipped": db_result.skipped,
+                "limit": req.paging_options.limit,
+                "total": db_result.total_count,
+                "page_number": req.paging_options.page_number,
+                "total_pages": calc_total_pages(db_result.total_count, req.paging_options.limit),
+                "found_records": db_result.uuids_and_types.len(),
+                "pubkey": db_result.pubkey
+            }));
```

Comment:
This breaks API backward compatibility as shown by the change you did to this test https://github.com/KomodoPlatform/komodo-defi-framework/blob/df102643bbe485cbaa422f1873af0a2ff127c9be/mm2src/mm2_test_helpers/src/for_tests.rs#L2428
But I see the problem since we can't apply paging in a right way to multiple dbs. Let's think more about a solution for this, otherwise we will have to make the query database/coin specific.

Reply:
what if we include `db_id` field in both `MakerSavedSwap` and `TakerSavedSwap` which will enable the joining of all results for db_ids in the rpc method above without storing in a vector..so the call will do the filtering using `db_id` field ?